libs/api/database/src/lib/content/stores/tags.store.ts

```diff
@@ -153,10 +172,20 @@ export class TagsStore {
             parent: form.parent,
         });
 
-        // Check that name not already in use
-        const tagWithSameName = await this.findTagByName(newTag.name)
-        if (tagWithSameName) {
-            throw new BadRequestException("There is already a tag with this name.");
+        // If child tag, check that name not already in use by its siblings
+        if (form.parent) {
+            const tagWithSameName = await this.findChildTagByNameAndParent(newTag.name, form.parent);
+            if (tagWithSameName) {
+                throw new BadRequestException("There is already a child tag of this parent with this name.");
+            }
+        }
+        // If parent tag, check that name not already in use by another parent tag
+        // Don't check for duplicates with children
+        else {
+            const tagWithSameName = await this.findParentTagByName(newTag.name)
+            if (tagWithSameName) {
+                throw new BadRequestException("There is already a parent tag with this name.");
+            }
```

Comment:
Improves name duplication check when creating/editing tags by just checking on the same level